packages/starlight/components/Hero.astro

```diff
@@ -50,12 +50,14 @@ if (image) {
 		{
 			actions.length > 0 && (
 				<div class="sl-flex actions">
-					{actions.map(({ attrs, icon, link: href, text, variant }) => (
-						<LinkButton {href} {variant} icon={icon?.name} {...attrs}>
-							{text}
-							{icon?.html && <Fragment set:html={icon.html} />}
-						</LinkButton>
-					))}
+					{actions.map(
+						({ attrs: { class: className, ...attrs } = {}, icon, link: href, text, variant }) => (
+							<LinkButton {href} {variant} icon={icon?.name} class:list={[className]} {...attrs}>
```

Comment:
Huh, interesting! Do you know why this extra work is needed? Because Astro already sets a `class` and uses that instead of the one spread in from `attrs`? Feels like maybe something we should report to the team.

Reply:
Good question, I always assumed this was the expected pattern to use with our `scopedStyleStrategy` when you want to accept a class that you would pass down to a child and the component also get a scoped class. Using `class` would result with one being overwritten by the other so you would have to use `class:list`. Thinking about it, I guess this may be handled automatically, never thought about it, could be a good question to ask 🤔

Reply:
Asked [on Discord](https://discord.com/channels/830184174198718474/1070208455329198190/1278042063937474605), but I think we can still make this as a fix for now!